server/storages/mysql.go

```diff
@@ -0,0 +1,254 @@
+	m.tableHelpers = []*TableHelper{tableHelper}
+	m.sqlAdapters = []adapters.SQLAdapter{adapter}
+	m.archiveLogger = config.loggerFactory.CreateStreamingArchiveLogger(config.destinationID)
+
+	//streaming worker (queue reading)
+	m.streamingWorker = newStreamingWorker(config.eventQueue, config.processor, m, tableHelper)
+	m.streamingWorker.start()
+
+	return m, nil
+}
+
+func (m *MySQL) DryRun(payload events.Event) ([]adapters.TableField, error) {
+	_, tableHelper := m.getAdapters()
+	return dryRun(payload, m.processor, tableHelper)
+}
+
+//Store process events and stores with storeTable() func
+//returns store result per table, failed events (group of events which are failed to process) and err
+func (m *MySQL) Store(fileName string, objects []map[string]interface{}, alreadyUploadedTables map[string]bool) (map[string]*StoreResult, *events.FailedEvents, error) {
+	_, tableHelper := m.getAdapters()
+	flatData, failedEvents, err := m.processor.ProcessEvents(fileName, objects, alreadyUploadedTables)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	//update cache with failed events
+	for _, failedEvent := range failedEvents.Events {
+		m.eventsCache.Error(m.IsCachingDisabled(), m.ID(), failedEvent.EventID, failedEvent.Error)
+	}
+
+	storeFailedEvents := true
+	tableResults := map[string]*StoreResult{}
+	for _, fdata := range flatData {
+		table := tableHelper.MapTableSchema(fdata.BatchHeader)
+		err := m.storeTable(fdata, table)
+		tableResults[table.Name] = &StoreResult{Err: err, RowsCount: fdata.GetPayloadLen(), EventsSrc: fdata.GetEventsPerSrc()}
+		if err != nil {
+			storeFailedEvents = false
+		}
+
+		//events cache
+		for _, object := range fdata.GetPayload() {
+			if err != nil {
+				m.eventsCache.Error(m.IsCachingDisabled(), m.ID(), m.uniqueIDField.Extract(object), err.Error())
+			} else {
+				m.eventsCache.Succeed(m.IsCachingDisabled(), m.ID(), m.uniqueIDField.Extract(object), object, table)
+			}
+		}
+	}
+
+	//store failed events to fallback only if other events have been inserted ok
+	if storeFailedEvents {
+		return tableResults, failedEvents, nil
+	}
+
+	return tableResults, nil, nil
+}
+
+//check table schema
+//and store data into one table
+func (m *MySQL) storeTable(fdata *schema.ProcessedFile, table *adapters.Table) error {
+	_, tableHelper := m.getAdapters()
+	dbSchema, err := tableHelper.EnsureTableWithoutCaching(m.ID(), table)
+	if err != nil {
+		return err
+	}
+
+	start := time.Now()
+	if err := m.adapter.BulkInsert(dbSchema, fdata.GetPayload()); err != nil {
+		return err
+	}
+	logging.Debugf("[%s] Inserted [%d] rows in [%.2f] seconds", m.ID(), len(fdata.GetPayload()), time.Now().Sub(start).Seconds())
+
+	return nil
+}
+
+//SyncStore is used in storing chunk of pulled data to Postgres with processing
+func (m *MySQL) SyncStore(overriddenDataSchema *schema.BatchHeader, objects []map[string]interface{}, timeIntervalValue string, cacheTable bool) error {
+	_, tableHelper := m.getAdapters()
```

Comment:
You can just use:

`return syncStoreImpl(m, overriddenDataSchema, objects, timeIntervalValue, cacheTable)`

Reply:
replaced with syncStoreImpl